lib/mix/lib/mix/tasks/compile.elixir.ex

```diff
@@ -79,6 +80,7 @@ defmodule Mix.Tasks.Compile.Elixir do
     docs: :boolean,
     warnings_as_errors: :boolean,
     ignore_module_conflict: :boolean,
+    undefined_variable_as_call: :boolean,
```

Comment:
I wouldn't add a command line option. I can't think of a scenario where we want to swap this on a single invocation.